framework/src/base/NonlinearEigenSystem.C

```diff
@@ -135,4 +145,14 @@ NonlinearEigenSystem::getNthConvergedEigenvalue(dof_id_type n)
   return _transient_sys.get_eigenpair(n);
 }
 
+void
+NonlinearEigenSystem::addKernelToExtraWarehouse(MooseSharedPointer<KernelBase> kernel, THREAD_ID tid)
```

Comment:
I'm not sure if I like the name of this routine "...ExtraWarehouse". Are the EigenKernels in the KernelWarehouse AND one of these other two warehouses too?